orchestrator/orchestrator.go

```diff
@@ -230,6 +233,20 @@ func (s *Orchestrator) resolveInput(wfHash hash.Hash, exec *execution.Execution,
 	return exec.Outputs.Fields[outputKey], nil
 }
 
+func (s *Orchestrator) resolveResult(wfHash hash.Hash, exec *execution.Execution, instanceHash hash.Hash, outputKey string) (*types.Value, error) {
+	if !wfHash.Equal(exec.ProcessHash) {
+		return nil, fmt.Errorf("reference's result nodeKey not found")
+	}
+	if !exec.InstanceHash.Equal(instanceHash) {
```

Comment:
i think this is not bad but you should also check that the `TaskKey` are the right one.